### PR TITLE
Create methods.md

### PR DESCRIPTION
**Pull request by zorggn**
FAQ covering colon and dot functional syntax.
Comments welcome. :3

### CODE REVIEW

**Comment by pfirsich** [comment]
Hey, thanks so much for taking the time to extend this FAQ! 
The first question I ask myself is "what exactly do you want to explain here?". I think the idea was to explain the colon syntax. Then why start out with some babble about CS words, links to wiki articles and stuff? Does someone that wants to know these things really need to know that functions are first class values? Will they even understand it? I'd drop the small remark about `_G` too. Know what you want to say and make a point of getting that across first and foremost. I think the `table.key` <-> `["key"]` is actually good example, because a beginner might have seen or understood that before and realizes that the `:`-syntax is just like that (in that it's only a simple replacement internally). I'd first explain what the colon does (i.e. show `self`-examples right away). I'm a big proponent of not EVER showing code examples that don't work :D (the erroneous one). People will copy-paste them. Swap the Function Call section to below the example section. And in the examples show a practical example. Like a simple class system or so (with closure based classes?). And I think in a document like this it is absolutely **mandatory** to link some class libraries.

**Comment by pablomayobre** [comment]
I would turn this into a "Syntax Sugar" guide on all the syntax sugar Lua has, and how to best use it.

With different sections for `self`, `.index` and `function name ()`

I agree that you shouldn't mention other things like first class functions, varargs, `__call` metamethod and all that stuff, specially if it isn't something you'll explain.

**Comment by zorggn** [comment]
I also included a part about the {} table constructor, since the line between what's syntax sugar and what's not kinda started to blur... or at the very least, i felt that the indexing might feel like a part of it was missing if i didn't include that.

**Comment by pfirsich** [comment]
I am really sorry this took so long!

This new one is really good, I like it a lot :)

I would like to have the examples be clearer about stuff being the same, e.g. I think a beginner might be confused about `tbl['anything']` and `tbl.something` and might think they are different things. Do something like this (probably not exactly) instead:

```lua
tbl = {}
tbl['anything'] = 'arbitrary string key'
print(tbl['anything'])
print(tbl.anything) -- the same as the above line
tbl['or']       = 'arbitrary string key that happens to be a lua keyword'
```

or `fun{a = 1, b = 2}` being the same as `fun({a = 1, b = 2})`

Also it would be perfect if you added some links to what you can do with the colon syntax (i.e. some oop tutorial) and maybe a link or two to class libraries.

**Comment by pfirsich** [comment]
I wanted to link this today. Would you consider making those adjustments? If you wouldn't I'd merge this also, since it's pretty good already, but I'd still like the changes tbh.

**Comment by zorggn** [comment]
Yes, sorry for taking too long, i'll go over it today one more time!

**Comment by zorggn** [comment]
I updated the file, rewrote the anything/something part to be a bit more clear... hopefully;
The () being optional in case of single table/string literals imo already explain it pretty clearly that e.g. `fun{whatever = 'in here'}` would be equivalent to `fun({whatever = 'in here'})`.

I also included a few class library links at the end.

**Comment by pfirsich** [comment]
Hey zorg, I didn't really look at it and don't plan to do so, but because I trust you and it was pretty good already, I'll just click "close and comment"! Sorry for all this hassle.

**Comment by pfirsich** [comment]
Yeah, should have clicked merge instead.

File: syntaxsugar.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Syntactic Sugar in Lua
-Syntax Sugar, in short, is a way to write something that might be more simpler to understand, or at least in an alternate way to how the language would express it by default. Lua has quite a few instances of this, which will be demonstrated below.
+Syntax Sugar, in short, is a way to write something in either a shorter way, or in a way that might be more simpler to understand to some, or at least in an alternate way to how the language would express it by default. Lua has quite a few instances of this, which will be demonstrated below.
```
Comment (pablomayobre) [inline syntaxsugar.md:2]: "more simpler" is redundant, just use "simpler"

Comment (zorggn) [reply syntaxsugar.md:2]: Completely agree... just need to find out what method would be the simplest and fastest to administer the change...

Comment (pfirsich) [reply syntaxsugar.md:2]: You can open another PR if you want to have your name on it, or I'll do it later!

Comment (pfirsich) [reply syntaxsugar.md:2]: I just rewrote that sentence, since I thought there is not much in it and it's a lot of words.

**Pull request merged**